pkg/ottl/ottlfuncs/func_delete_key.go

```diff
@@ -17,23 +17,35 @@ package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-c
 import (
 	"context"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
-func DeleteKey[K any](target ottl.Getter[K], key string) (ottl.ExprFunc[K], error) {
+type DeleteKeyArguments[K any] struct {
+	Target ottl.Getter[K] `arg:"0"`
```

Comment:
Is there any chance that the `arg` tag name will conflict with any existing popular tags? I want to make sure that our required tag is usable if someone makes a custom function that also uses tags to do something else.

Reply:
I was operating under the assumption that nobody would have an interest in parsing the tags on these structs, but it's also fair to say that we should make room for that. I've updated it to `ottlarg`, which I think is still short enough not to be too noisy.